_posts/2022-01-19-7.0.0-release.md

```diff
@@ -0,0 +1,227 @@
+(ARROW-14181).
+
+Residual predicates have been implemented in the hash join node
+(ARROW-13643).
+
+The "list_parent_indices" function now always returns int64 data
+regardless of the input type (ARROW-14592).
+
+Month-day-nano interval data is now supported as input of the same functions
+as other interval types (ARROW-13989).
+
+### CSV
+
+The CSV writer got additional configuration options:
+- the string representation of null values (ARROW-14905);
+- the quoting strategy: always / never / as needed (ARROW-14905);
+- the end of line character(s) (ARROW-14907)
+
+### Dataset Layer
+
+"SkyHook", a dataset addition that offloads fragment scan operations to a
+Ceph distributed storage, was contributed (ARROW-13607).
+
+The dataset writer now exposes options `min_rows_per_group` and
+`max_rows_per_group` to control the size of row groups created (ARROW-14426).
+
+### IO and Filesystem Layer
+
+A critical bug in the AWS SDK for C++ that risks losing data in S3 multipart
+uploads has been circumvented (ARROW-14523).
+
+The Google Cloud Storage filesystem is now featureful enough to pass all
+generic filesystem tests (ARROW-14924).
+
+The OpenAppendStream method of filesystems has been un-deprecated; however,
+it still cannot be implemented for all filesystem backends (ARROW-14969).
+
+A new function `arrow::fs::ResolveS3BucketRegion` allows resolving the
+region where a particular S3 bucket resides (ARROW-15165).
+
+The S3 filesystem now sets the Content-Type of output files to
+"application/octet-stream" (instead of "application/xml" previously)
+if not explicitly specified by the caller (ARROW-15306).
+
+### IPC
+
+Fine-grained I/O (coalescing) is now enabled in the synchronous (ARROW-12683)
+and asynchronous (ARROW-14577) IPC reader.
+
+It is now possible to set the compression level when using LZ4 compression
+(ARROW-9648).
+
+### Parquet
+
+DELTA_BYTE_ARRAY-encoded data can now be read from (but not written to)
+bytearray columns in Parquet files (PARQUET-492).
+
+## C# notes
+
+TBD
+
+## Go notes
+
+### Bug Fixes
+
+TBD
+
+### Enhancements
+
+TBD
+
+## Java notes
+
+TBD
+
+## JavaScript notes
+
+TBD
+
+## Python notes
+
+TBD
+
+## R notes
+
+TBD
+
+## Ruby and C GLib notes
+
+### Ruby
+
+There are two new contributors @okadakk and @simpl1g .
+
+The updates of Red Arrow consists of the following improvements:
+- Addition of a [cookbook section](https://github.com/apache/arrow/pull/11584) to the [main repository](https://github.com/apache/arrow/tree/master/ruby)
+- Improved [Arrow::Function#execute useability](https://github.com/apache/arrow/pull/12101)
+- Support for [table joins](https://github.com/apache/arrow/pull/12108)
+- A [MacOS Development guide](https://github.com/apache/arrow/pull/11181)
+- Enable separate use of [min and max](https://github.com/apache/arrow/pull/11179)
+- Add support for loading [Arrow::Table from a URI](https://github.com/apache/arrow/pull/12128)
```

Comment:
```suggestion
### Ruby

There are two new contributors @okadakk and @simpl1g .

The updates of Red Arrow consists of the following improvements:

- `Arrow::Function#execute` now accepts an instance of an `Arrow::Column` as its argument [(ARROW-14551)](https://issues.apache.org/jira/browse/ARROW-14551)
- `Arrow::TableLoader` now supports `.arrows` files to load [(ARROW-15356)](https://issues.apache.org/jira/browse/ARROW-15356)
- Add support for loading `Arrow::Table` from URI in `Arrow::TableLoader` [(ARROW-14562)](https://issues.apache.org/jira/browse/ARROW-14562)
- `Arrow::Table` now supports to join two tables [(ARROW-14531)](https://issues.apache.org/jira/browse/ARROW-14531)
- `Arrow::Function#execute` gets more easier to use than before [(ARROW-15274)](https://issues.apache.org/jira/browse/ARROW-15274)
- `Arrow::SortKey#name` has been renamed to `Arrow::SortKey#target` [(ARROW-14784)](https://issues.apache.org/jira/browse/ARROW-14784)
- Add Cookbook section to documentation [(ARROW-14636)](https://issues.apache.org/jira/browse/ARROW-14636)
- Support the explicit initialization of S3 API by the `Arrow.s3_initialize` method [(ARROW-14637)](https://issues.apache.org/jira/browse/ARROW-14637)
- On macOS, stop specifying the version of openssl package explicitly when building the extension library [(ARROW-14619)](https://issues.apache.org/jira/browse/ARROW-14619)
```

Reply:
@kou Could you check this change?

Reply:
Thanks!

> - Add support for loading `Arrow::Table` from URI in `Arrow::TableLoader` [(ARROW-14562)](https://issues.apache.org/jira/browse/ARROW-14562)

How about "Add support for `Arrow::Table.load(URI("https:///.../xxx.arrow"))`" or something? `Arrow::TableLoader` is an internal class. It's not for users.

Reply:
Updated. Thank you.